contextproject-ms3/src/main/java/contextproject/models/Key.java

```diff
@@ -1,166 +1,181 @@
 package contextproject.models;
 
-import com.mpatric.mp3agic.Mp3File;
-
 import java.util.ArrayList;
-
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class Key {
-  private static Logger log = LogManager.getLogger(Key.class.getName());
-
-  private int keyNumberPart;
-  private String keyCharPart;
-  private String keyMusicNotation;
+  private int normalizedKeyNumber;
+  private String normalizedKeyFlag;
+  private String musicalKeyString;
 
   /**
    * Get the key of a song from the meta data.
    * 
-   * @param song
-   *          song to get key from
+   * @param keyString String key from ID3 information
    */
-  public Key(Mp3File song) {
-    if (song.hasId3v2Tag()) {
-      keyMusicNotation = song.getId3v2Tag().getKey();
-      if(keyMusicNotation != null) {
-        convertKey();
-      }
-      log.info(song.getFilename() + " has no Id3v2Tag");
+  public Key(String keyString) throws IllegalArgumentException {
+    if (keyString != null) {
+      normalizeKey(keyString);
+    } else {
+      throw new IllegalArgumentException("Key should not be null");
```

Comment:
This throw specifically was meant to let the Track class know that something went wrong, which is also why Key throws IllegalArgumentException. The Track class catches this and logs it. :)